diff --git a/dist/client/overlay.d.mts b/dist/client/overlay.d.mts
index 221f83fa7d298ce45353ff77e9711e1ceed80825..37cdbf96ce12e3740a42f29073c7ab0e7eaa9cc0 100644
--- a/dist/client/overlay.d.mts
+++ b/dist/client/overlay.d.mts
@@ -29,6 +29,17 @@ declare const state: {
         getParent: () => ElementTraceInfo | undefined;
         getElementsSamePosition: () => Element[] | undefined;
     } | undefined;
+    mainRect?: {
+        height: number;
+        width: number;
+        x: number;
+        y: number;
+        readonly bottom: number;
+        readonly left: number;
+        readonly right: number;
+        readonly top: number;
+        toJSON: () => any;
+    } | undefined;
     sub: {
         rects?: {
             id: string;
diff --git a/dist/client/overlay.mjs b/dist/client/overlay.mjs
index 99dc0fbf62db5798490cdc15e3dc7d4f5b6f6b43..66c8f5acbd3f6bc12172341553be7ea75c850036 100644
--- a/dist/client/overlay.mjs
+++ b/dist/client/overlay.mjs
@@ -77,8 +77,8 @@ function createOverlay() {
     overlay.style.opacity = state.isVisible ? "1" : "0";
   });
   watchEffect(() => {
-    if (state.main && state.main.rect) {
-      const rect = state.main.rect;
+    if (state.main && state.mainRect) {
+      const rect = state.mainRect;
       mainRect.style.opacity = "1";
       if (state.isFocused) {
         mainRect.setAttribute("rx", "8");
@@ -164,10 +164,12 @@ function update(result) {
     state.isVisible = false;
     state.sub.rects = void 0;
     state.main = void 0;
+    state.mainRect = void 0;
     return;
   }
   state.isVisible = true;
   state.main = result;
+  state.mainRect = result.rect;
   const samePos = result.getElementsSamePosition();
   state.sub.rects = samePos?.map((el) => ({
     id: getElementId(el),
